dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/trackedentity/TrackerOwnershipManagerTest.java

```diff
@@ -136,4 +143,20 @@ void testTransferOwnership()
         assertFalse( trackerOwnershipAccessManager.hasAccess( userA, entityInstanceA1, programA ) );
         assertTrue( trackerOwnershipAccessManager.hasAccess( userB, entityInstanceA1, programA ) );
     }
+
+    @Test
+    void testAccessToClosedProgramWithoutOwnershipOnlyIfUserInSearchScope()
+    {
+        assertTrue( trackerOwnershipAccessManager.hasAccess( userA, entityInstanceA1, programB ) );
```

Comment:
Its hard to understand why one should be true and the other false just from reading the test 😅 It's often difficult with how our tests are structured.

Is the reason why userB does not have access because its a closed program?
userA has access because programB is in search scope right? Where is that set?

IMHO I like to see these important bits in the test. So a little setup that shows these.

If you wanted we could pair on it to see if there is a way to structure the tests differently 😄

Reply:
I refactored this and now I'm testing one thing at a time with a more explicit naming, so hopefully it's a bit easier to understand what each test does. If you think there's still room for significant improvement, I'd be happy to pair program and find a better solution.